### PR TITLE
Download Button for Plots

### PR DESCRIPTION
**Pull request by Konkaseeti**
Download Plot button added to Coral Health Explorer > Dead Coral.

Changed the Plot render function, It is now first created as a reactive object so that ggsave can utilize it,

Then passed to renderPlot() which will still place the plot on the UI.

Download Handler function added for the plot I was working on as well.

### CODE REVIEW

File: server.R [hunk 1]
```diff
@@ -15,7 +15,7 @@ shinyServer(function(input, output) {
     coral_health_plot_caption <- reactive({
         generate_coral_health_caption(input)
     })
-    output$coral_health_plot <- renderPlot({
+    coral_health_plot <- reactive({
```
Comment (sharp-lilac) [inline server.R:18]: Good solution - this works well, and the app still runs fast with the reactive expression

File: server.R [hunk 2]
```diff
@@ -225,4 +228,13 @@ shinyServer(function(input, output) {
...
     )
+
+    output$deadCoralDl <- downloadHandler(
+        filename = function() {
+            paste("coral_health_plot", Sys.Date(), ".png", sep = "")
+        },
+        content = function(file) {
+            ggsave(file, plot = coral_health_plot(), width = 15, height = 15)
+        }
+    )
```
Comment (sharp-lilac) [inline server.R:239]: Perfect, great!

File: ui.R [hunk 3]
```diff
@@ -145,7 +145,8 @@ ui <- dashboardPage(
                                     outline = TRUE,
                                     status = "primary",
                                     icon = icon("check")
-                                )
+                                ),
+                                downloadButton("deadCoralDl", "Download Plot")
```
Comment (sharp-lilac) [inline ui.R:149]: Could you please name this ID based on 'snake case,' which an internal styling standard we have at ERI? All undercase, separated words by underscores. For example:
`coral_health_plot_download`
Once you change it here, you will also have to change it in the server too to match. Thanks!

File: server.R [hunk 4]
```diff
@@ -225,4 +228,13 @@ shinyServer(function(input, output) {
             file.copy("www/images/Turneffe_Map.jpg", file)
         }
     )
+
```
Comment (sharp-lilac) [inline server.R:231]: Please add a section title, for example:
`# Download coral health plot`

**Review by sharp-lilac — CHANGES_REQUESTED** [review]
Nice work! Just a few edits. I appreciate your contributions to the project

**Comment by Konkaseeti** [comment]
Did the mentioned edits. Commenting incase.

**Comment by sharp-lilac** [comment]
@Konkaseeti Did you commit the changes and push? They aren't showing for me.

**Comment by Konkaseeti** [comment]
I did, I cant make another commit without making a change

**Review by sharp-lilac — APPROVED** [review]
Great! Thank you, found the changes

**Pull request merged**